modules/customize-certificates-add-service-serving-crd.adoc

```diff
@@ -0,0 +1,62 @@
+// Module included in the following assemblies:
+//
+// * authentication/certificates/service-serving-certificate.adoc
+
+[id="add-service-certificate-crd_{context}"]
+= Add a service certificate to a Custom Resource Definition
+
+////
+
+// TODO: intro + important note?
+
+A Pod can access the service CA certificate by mounting a ConfigMap that
+is annotated with `service.beta.openshift.io/inject-cabundle=true`.
+Once annotated, the cluster automatically injects the service CA
+certificate into the `service-ca.crt` key on the ConfigMap. Access to
+this CA certificate allows TLS clients to verify connections to
+services using service serving certificates.
+
+[IMPORTANT]
+====
+After adding this annotation to a ConfigMap all existing data in it is
+deleted. It is recommended to use a separate ConfigMap to contain the
+`service-ca.crt`, instead of using the same ConfigMap that stores your
+Pod's configuration.
+====
+////
+
+.Procedure
+
+. Annotate the CRD with `service.beta.openshift.io/inject-cabundle=true`.
++
+----
+$ oc annotate crd <crd-name> \//<1>
+     service.beta.openshift.io/inject-cabundle=true
+----
+<1> Replace `<crd-name>` with the name of the CRD to annotate.
++
+////
+TODO: Need this note for CRDs?
+[NOTE]
+====
+Explicitly referencing the `service-ca.crt` key in a volumeMount will
+prevent a Pod from starting until the ConfigMap has been injected with
+the CA bundle.
+====
+////
++
+For instance, to annotate the CRD `foo` the following command would be
+used:
++
+----
+$ oc annotate crd foo service.beta.openshift.io/inject-cabundle=true
+----
+
+. View the CRD to ensure the certificate has been generated. This
+appears as a `service-ca.crt` in the YAML output.
++
+----
+$ oc get crd <crd-name> -o yaml
+
+TODO: need output
```

Comment:
@marun I either need some appropriate output from you, or how to test setting up the CRD to have the webhook for conversion.

Reply:
@marun And also, suggestions for what you what we ought to add here about that requirement ^

Reply:
I think documenting configuration of webhook conversion here is out-of-scope for this doc. The only part that service ca plays is in injecting the ca bundle when using a webhook for conversion that is secured with a service ca certificate. If there is openshift documentation for configuring webhook conversion for a CRD, link to that. If not, can we link to upstream docs?